File: Ruby_2Darbas/CIKLAI/Ruby/antras.rb
# frozen_string_literal: true

def DvigubasSkaitmenys(a)
  sk2 = 0
  sk3 = 0
  i = 0
  sk1 = a.to_i
  while sk1 != 0
    sk3 = 10 * sk3.to_i + sk1.to_i % 10
    sk1 /= 10
    i += 1
  end
  puts 'skaicius su padvigubintais skaitmenimis:'
  while i.positive?
    sk2 = sk2 * 100 + (sk3 % 10) * 10 + (sk3 % 10)
    sk3 /= 10
    i -= 1
  end
  #puts sk2
  return sk2
end

File: Ruby_2Darbas/CIKLAI/antras.cpp
#include <iostream>

using namespace std;

int main()
{
    int sk1=0,sk2=0,a=0,sk3=0,b;
    int i = 0;
    cout<<"Iveskite skaiciu"<<endl;
    cin>>a;
    sk1=a;
    do //apvercia skaiciu
    {
        sk3=10*sk3+sk1%10;
        sk1=sk1/10;
        i++;
    }   while (sk1!=0);

    cout<<"skaicius su padvigubintais skaitmenimis:"<<endl;
    while (i > 0) //spausdina dviguba atvirksciai
    {
        sk2=sk2*100+(sk3%10)*10+(sk3%10);
        sk3=sk3/10;
        i--;
    }
        cout<<sk2<<endl;
return 0;
}
